core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp

```diff
@@ -121,10 +116,6 @@ int cuda_get_max_block_size(const CudaInternal* cuda_instance,
         opt_threads_per_sm = threads_per_sm;
       }
     }
-    // printf("BlockSizeMax: %i Shmem: %i %i %i %i Regs: %i %i Blocks: %i %i
-    // Achieved: %i %i Opt: %i %i\n",block_size,
-    //   shmem_per_sm,max_shmem_per_block,functor_shmem,total_shmem,
-    //   regs_per_sm,regs_per_thread,max_blocks_shmem,max_blocks_regs,blocks_per_sm,threads_per_sm,opt_block_size,opt_threads_per_sm);
```

Comment:
For one these print statement are useful to debug, but also they will conflict with #3103

Reply:
If they're useful for debugging, they shouldn't be commented out, but surrounded with `#ifdef KOKKOS_DEBUG` or something like that (or `#ifdef KOKKOS_IMPL_CUDA_BLOCK_SIZE_DEBUG` if that is preferable). Letting code just rot behind a comment isn't helpful, though.

Reply:
I agree with everything you said.  I would prefer if you did not touch that file but I can resolve conflicts later.

Reply:
Okay I reverted it